assets/src/components/v2/gl_eink_double/line_map.tsx

```diff
@@ -24,6 +25,22 @@ const abbreviateStop = (stop) => {
   return stop;
 };
 
+// This function splits the label into two parts as close to the middle as possible.
+// Will only split on a space.
+const splitLabel = (labelText: string) => {
+  let middle = Math.floor(labelText.length / 2);
+  let before = labelText.lastIndexOf(" ", middle);
+  let after = labelText.indexOf(" ", middle + 1);
+
+  if (middle - before < after - middle) {
+    middle = before;
+  } else {
+    middle = after;
+  }
+
+  return [labelText.substring(0, middle), labelText.substring(middle + 1)];
```

Comment:
Will we always find a space character? If one isn't found on either side of `middle` then this returns `["", labelText]`.

I did some checking of the possible results of this on paper and overall it looks right, but you might want to explicitly handle cases where a space isn't found on the left or right side, or both, just to be safe.